src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/PrisonerIndexResource_compareIndexTest.kt

```diff
@@ -83,56 +86,19 @@ class PrisonerIndexResource_compareIndexTest : AbstractSearchDataIntegrationTest
 
   @Test
   fun `Reconciliation - no differences`() {
-    prisonMockServer.stubFor(
-      WireMock.get(WireMock.urlEqualTo("/api/offenders/ids"))
-        .willReturn(
-          WireMock.aResponse()
-            .withHeader("Content-Type", "application/json")
-            .withHeader("Total-Records", "10")
-            .withBody(
-              """[
-                { "offenderNumber": "A9999AA" },
-                { "offenderNumber": "A9999AB" },
-                { "offenderNumber": "A9999AC" },
-                { "offenderNumber": "A9999RA" },
-                { "offenderNumber": "A9999RB" },
-                { "offenderNumber": "A9999RC" },
-                { "offenderNumber": "A7089EY" },
-                { "offenderNumber": "A7089EZ" },
-                { "offenderNumber": "A7089FA" }]""",
-            ),
-        ),
-    )
     webTestClient.get().uri("/prisoner-index/reconcile-index")
       .headers(setAuthorisation(roles = listOf("ROLE_PRISONER_INDEX")))
       .exchange()
       .expectStatus().isAccepted
 
+    await untilCallTo { getNumberOfMessagesCurrentlyOnIndexQueue() } matches { it!! > 0 }
```

Comment:
this looks flaky, can't number of message transition from a positive number to zero quickly? In which case the await on line 94 will never work?

Reply:
the endpoint returns very fast and it takes a while for the queues to build up so should be ok. It was more likely that the queue was found to be empty at this point because the build up of messages had not started